### PR TITLE
Fixed Thrift exit verbosity in Windows

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by PoppySeedPlehzr** [comment]
On windows, when exiting the osqueryi client, there's some verbosity from Thrift being output. This is because Thrift just writes it's logging information directly to console. I spoke with @yying about this issue and he pointed me to the concept of changing where thrift writes it's output as mentioned here: `http://stackoverflow.com/questions/32066782/where-does-the-apache-thrift-c-globaloutput-output-to`

Before:
```
C:\Users\Nick\Documents\GitHub\poppy-osquery [master ≡]> .\build\windows10\osquery\Release\osqueryi.exe
osquery - being built, with love, at Facebook
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Using a virtual database. Need help, type '.help'
osquery> .quit
Thrift: Wed Sep 21 13:54:40 2016 TPipeServer ConnectNamedPipe GLE=errno = 995
Thrift: Wed Sep 21 13:54:40 2016 TServerTransport died: TPipeServer: client connection failed
```

After:
```
C:\Users\Nick\Documents\GitHub\poppy-osquery [windows-thrift-verbosity]> .\build\windows10\osquery\Release\osqueryi.exe
osquery - being built, with love, at Facebook
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Using a virtual database. Need help, type '.help'
osquery> .quit
C:\Users\Nick\Documents\GitHub\poppy-osquery [windows-thrift-verbosity]>
```

File: osquery/extensions/interface.cpp [hunk 1]
```diff
@@ -64,6 +67,23 @@ void ExtensionHandler::shutdown() {
...
+ * instead of the default output for thrift.
+ *
+ * @param msg The text to be logged
+ */
+void thriftLoggingOutput(const char* msg) {
+  VLOG(1) << msg;
+}
+
+ExtensionManagerHandler::ExtensionManagerHandler() {
+  GlobalOutput.setOutputFunction(thriftLoggingOutput);
```
Comment (theopolis) [inline osquery/extensions/interface.cpp]: This is awesome, is there any tear down we need to handle (as in when the process exits)?

File: osquery/extensions/interface.cpp [hunk 2]
```diff
@@ -64,6 +67,23 @@ void ExtensionHandler::shutdown() {
...
+ * @brief Updates the Thrift server output to be VLOG
+ *
+ * On Windows, the thrift server will output to stdout, which displays
+ * messages to the user on exiting the client. This function is used
+ * instead of the default output for thrift.
+ *
+ * @param msg The text to be logged
+ */
+void thriftLoggingOutput(const char* msg) {
+  VLOG(1) << msg;
```
Comment (theopolis) [inline osquery/extensions/interface.cpp]: Maybe: `VLOG(1) << "Thrift message: " << msg;` so we can easily filter them with simple expressions later on.

**Comment by facebook-github-bot-2** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2500/files/fc19d74d6fcc7d0606b325ae13e5d91d1a7e7953..a80044228be330fb167e535fe2cca57f482e69ae)

**Comment by osqueryer** [comment]
:-1: The commit a80044228be330fb167e535fe2cca57f482e69ae (Job results: [3304](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3304/)) failed one or more tests (**OS X/Linux**).

**Comment by PoppySeedPlehzr** [comment]
Fail results:

Centos 7
```
/usr/local/osquery/bin/ld: final link failed: No space left on device
clang-3.8: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [osquery/osquery_tables_tests] Error 1
make[2]: *** [osquery/CMakeFiles/osquery_tables_tests.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [all] Error 2
```

Centos 6
```
4/9 Test #4: python_test_osqueryi .............***Failed    0.07 sec
Cannot import thrift: pip install thrift?

    Start 5: python_test_osqueryd
5/9 Test #5: python_test_osqueryd .............***Failed    0.06 sec
Cannot import thrift: pip install thrift?

    Start 6: python_test_modules
6/9 Test #6: python_test_modules ..............***Failed    0.06 sec
Cannot import thrift: pip install thrift?

    Start 7: python_test_extensions
7/9 Test #7: python_test_extensions ...........***Failed    0.06 sec
Cannot import thrift: pip install thrift?

    Start 8: python_test_additional
8/9 Test #8: python_test_additional ...........***Failed    0.06 sec
Cannot import thrift: pip install thrift?

    Start 9: python_test_example_queries
9/9 Test #9: python_test_example_queries ......***Failed    0.06 sec
Cannot import thrift: pip install thrift?
```

**Comment by theopolis** [comment]
ok to test

**Comment by theopolis** [comment]
ok to test

**Pull request merged**